Sources/DiagnosticsLogger.swift

```diff
@@ -15,10 +15,13 @@ public final class DiagnosticsLogger {
 
     static let standard = DiagnosticsLogger()
 
-    private lazy var location: URL = FileManager.default.documentsDirectory.appendingPathComponent("diagnostics_log.txt")
+    private lazy var logFileLocation: URL = FileManager.default.documentsDirectory.appendingPathComponent("diagnostics_log.txt")
+    private var logFileHandle: FileHandle?
+
     private let inputPipe: Pipe = Pipe()
     private let outputPipe: Pipe = Pipe()
-    private let queue: DispatchQueue = DispatchQueue(label: "com.wetransfer.diagnostics.logger", qos: .utility, target: .global(qos: .utility))
+
+    private let queue: DispatchQueue = DispatchQueue(label: "com.wetransfer.diagnostics.logger", qos: .utility, autoreleaseFrequency: .workItem, target: .global(qos: .utility))
```

Comment:
Great, I did not know about `autoreleaseFrequency` so I learned something new! Just out of curiosity, do you know the difference between `qos` and `targetQueue` parameter?

Reply:
By using `targetQueue` you basically reuse and existing available queue with the same quality of service and dispatch to that linked thread. Otherwise, a new queue is generated with its own Thread. You can use this to minimize the total number of threads while still preserving the required execution semantics.

Reply:
We should, btw, do this way more often in our app! I'll see if I can do that as it's not that hard

Reply:
Thanks for explaining, that makes sense. Although I think it really depends on the context in which the dispatch queue is used. I need to refresh my knowledge on this but I remember from this [WWDC talk on GCD](https://developer.apple.com/videos/play/wwdc2017/706/)  that dispatching a lot of different task in your app to the same global queue can lead to similar issues with having too many threads as you can run into when you create many custom queues in your app.

Reply:
Yeah, it's worth spending some time on those talks for sure!